common/models/user.js

```diff
@@ -544,7 +544,11 @@ module.exports = function(User) {
     this.settings.ttl = this.settings.ttl || DEFAULT_TTL;
 
     UserModel.setter.password = function(plain) {
-      if (plain.indexOf('$2a$') === 0 && plain.length === 60) {
+      var indexOf = String.prototype.indexOf;
+      if (!plain) {
+        return;
+      }
+      if (indexOf.call(plain, '$2a$') === 0 && plain.length === 60) {
```

Comment:
#1437 states that if plain is type number indexOf breaks as it a method on string instances. Using indexOf.call keeps this from breaking on numbers as it will cast it to string first.